async-embedded/Cargo.toml

```
@@ -7,18 +7,20 @@ publish = false
 version = "0.0.0-alpha.0"
 
 [dependencies]
-cortex-m = { version = "0.6.2", optional = true }
-cortex-m-udf = { path = "../cortex-m-udf", optional = true }
 generic-array = "0.13.2"
 heapless = { git = "https://github.com/japaric/heapless", branch = "slab" }
 pin-utils = "0.1.0-alpha.4"
-riscv = { version = "0.5.6", optional = true }
 typenum = "1.11.2"
 
+[target.'cfg(target_arch = "arm")'.dependencies]
+cortex-m = "0.6.2"
+cortex-m-udf = { path = "../cortex-m-udf" }
+
+[target.'cfg(target_arch = "riscv32")'.dependencies]
```

```suggestion
[target.'cfg(any(target_arch = "riscv32", target_arch = "riscv64"))'.dependencies]
```

Maybe it's a good idea to add riscv64 too :)

---

Done. No way for me to test it though

---

Nice catch.

I wanted to keep the `wait-extern` ISA-agnostic, since other ISA's might need a similar backup soltion. But let's keep this PR focused on RISC-V.